libs/libc/locale/lib_iconv.c

```diff
@@ -0,0 +1,1423 @@
+ * Private Data
+ ****************************************************************************/
+
+/* Definitions of g_charmaps. Each charmap consists of:
+ * 1. Empty-string-terminated list of null-terminated aliases.
+ * 2. Special type code or number of elided quads of entries.
+ * 3. Character table (size determined by field 2), consisting
+ *    of 5 bytes for every 4 characters, interpreted as 10-bit
+ *    indices into the g_legacy_chars table.
+ */
+
+static const unsigned char g_charmaps[] =
+{
+  "utf8\0char\0\0\310"
+  "wchart\0\0\306"
+  "ucs2be\0\0\304"
+  "ucs2le\0\0\305"
+  "utf16be\0\0\302"
+  "utf16le\0\0\301"
+  "ucs4be\0utf32be\0\0\300"
+  "ucs4le\0utf32le\0\0\303"
+  "ascii\0usascii\0iso646\0iso646us\0\0\307"
+  "utf16\0\0\312"
+  "ucs4\0utf32\0\0\313"
+  "ucs2\0\0\314"
+#ifdef CONFIG_LIBC_LOCALE_JAPANESE
+  "eucjp\0\0\320"
+  "shiftjis\0sjis\0\0\321"
+  "iso2022jp\0\0\322"
+#endif
+#ifdef CONFIG_LIBC_LOCALE_CHINESE
+  "gb18030\0\0\330"
+  "gbk\0\0\331"
+  "gb2312\0\0\332"
+  "big5\0bigfive\0cp950\0big5hkscs\0\0\340"
+#endif
+#ifdef CONFIG_LIBC_LOCALE_KOREAN
+  "euckr\0ksc5601\0ksx1001\0cp949\0\0\350"
+#endif
+#ifdef CONFIG_LIBC_LOCALE_CODEPAGES
+  #include "codepages.h"
+#endif
+};
+
+/* Table of characters that appear in legacy 8-bit codepages,
+ * limited to 1024 slots (10 bit indices). The first 256 entries
+ * are elided since those characters are obviously all included.
+ */
+
+static const unsigned short g_legacy_chars[] =
+{
+  #include "legacychars.h"
+};
+
+#ifdef CONFIG_LIBC_LOCALE_JAPANESE
+static const unsigned short g_jis0208[84][94] =
+{
+  #include "jis0208.h"
+};
+
+static const unsigned short g_rev_jis[] =
+{
+  #include "revjis.h"
+};
+#endif
+
+#ifdef CONFIG_LIBC_LOCALE_CHINESE
+static const unsigned short g_gb18030[126][190] =
+{
+  #include "gb18030.h"
+};
+
+static const unsigned short g_big5[89][157] =
+{
+  #include "big5.h"
+};
+
+static const unsigned short g_hkscs[] =
+{
+  #include "hkscs.h"
+};
+#endif
+
+#ifdef CONFIG_LIBC_LOCALE_KOREAN
+static const unsigned short g_ksc[93][94] =
+{
+  #include "ksc.h"
+};
+#endif
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static int fuzzycmp(FAR const unsigned char *a, FAR const unsigned char *b)
+{
+  for (; *a && *b; a++, b++)
+    {
+      while (*a && (*a | 32U) - 'a' > 26 && *a - '0' > 10U)
+        a++;
+
+      if ((*a | 32U) != *b)
+        {
+          return 1;
+        }
+    }
+
+  return *a != *b;
+}
+
+static size_t find_charmap(FAR const void *name)
+{
+  FAR const unsigned char *s;
+
+  if (*(FAR char *)name == 0)
+    {
+      /* "utf8" */
+
+      name = g_charmaps;
+    }
+
+  for (s = g_charmaps; *s; )
+    {
+      if (!fuzzycmp(name, s))
+        {
+          for (; *s; s += strlen((FAR void *)s) + 1);
+          return s + 1 - g_charmaps;
+        }
+
+      s += strlen((FAR void *)s)+1;
```

Comment:
```suggestion
      s += strlen((FAR void *)s) + 1;
```
why cast to `void *`?

Reply:
Done!